Wikipedia/Code/TalkPagesFunnel.swift

```diff
@@ -0,0 +1,87 @@
+import Foundation
+
+final class TalkPagesFunnel: EventLoggingFunnel, EventLoggingStandardEventProviding {
+   
+    public static let shared = TalkPagesFunnel()
+    
+    private enum Action: String, Codable {
+        case openTopic = "open_topic"
+        case newTopicClick = "new_topic_click"
+        case replyClick = "reply_click"
+        case refresh
+        case langChange = "lang_change"
+        case submit
+    }
+    
+    private enum Source: String, Codable {
+        case talkPage = "talk_page"
+        case article
+        case notificationsCenter = "notifications_center"
+        case deepLink = "deep_link"
+        case account
+        case search
+        case inAppWebView = "in_app_web_view"
+        case unknown
+        
+        init(routingSource: RoutingUserInfoSourceValue) {
+            switch routingSource {
+            case .talkPage: self = .talkPage
+            case .article: self = .article
+            case .notificationsCenter: self = .notificationsCenter
+            case .deepLink: self = .deepLink
+            case .account: self = .account
+            case .search: self = .search
+            case .inAppWebView: self = .inAppWebView
+            case .unknown: self = .unknown
+            }
+        }
+    }
+    
+    private struct Event: EventInterface {
+        static let schema: EventPlatformClient.Schema = .talkPages
+        let action: Action
+        let source: Source
+        let page_ns: String
+        let time_spent: Int
+        let wiki_db: String
```

Comment:
Just noting that this prop was changed to wiki_id as suggested on the schema

Reply:
Done!